dd

kadirkidir opened the pull request:
dd

KarByc committed:
[VSE] Added default ego prefab to ego vehicles without mesh renderers.

lokesku committed:
Make controllables configurable on jenkins

kamkolak-lge committed:
Fix TF visual and sensor hierarchy

lokesku committed:
Add new FORCE_DEV_BUILD param to jenkins

KarByc committed:
Added public driver view transform to be set in the Inspector.

EricBoiseLGSVL committed:
update input for g920 wheel for windows

KarByc committed:
Loads assembly with the ego name from asset bundle.

lokesku committed:
Add Obstacle to NPC layer mask

KarByc committed:
[Network] Distributing all sensors.

Setting IsAuthoritative dynamically.
Messages forwarding support.

EricBoiseLGSVL committed:
[FEATURE] [AUTO-6521] Ped build bundle

EricBoiseLGSVL committed:
Support Bicycle npc and remove core npc assets

kamkolak-lge committed:
Fix overlapping rain chunks

daviduhm committed:
AUTO-6422: Publish camera intrinsic parameters to ROS and ROS2

mark-gerow-lge committed:
Pedestrian callbacks added

EricBoiseLGSVL committed:
small change to prevent walker ped from gliding

shr-project committed:
AUTO-6394: Jenkinsfile*: use dockerhub-docker-registry/auto-gitlab-docker-registry credentials

* to login to both auto-gitlab as well as default docker registry (hub.docker.com)

kamkolak-lge committed:
[AUTO-6410] Update Unity to 2020.3.3f1

KarByc committed:
[VSE] Loading pedestrians from config.

EricBoiseLGSVL committed:
Rotate sun and moon to match map rotation

KarByc committed:
[Network] Distributed simulation can be run on a single machine with multiple Simulator instances.

Fixed stopping sensors download process.

ericSterner committed:
[Fix] [AUTO-6662] Unity upgrade: lights in scenes fixed values

EricBoiseLGSVL committed:
get full path from data path provided

KarByc committed:
[VSE] Fixed changing agents' variants.

ericSterner committed:
[Fix] [AUTO-6665] Fix brown lighting

EricBoiseLGSVL committed:
add missing exposure change to max

shr-project committed:
AUTO-6719: Jenkins/README.md: fix typos and variable name

* DOCKER_IMAGE_NAME should be SIMULATOR_DOCKER_IMAGE

shr-project committed:
AUTO-6719: Jenkins/README.md: export FORCE_DEV_BUILD and CODE_SIGNING_FILE as well

* without CODE_SIGNING_FILE docker-compose will fail with:
  ERROR: Cannot create container for service build-simulator: create .: volume name is too short, names should be at least two alphanumeric characters

* and without FORCE_DEV_BUILD:
  /mnt/Jenkins/build-simulator.sh: line 60: FORCE_DEV_BUILD: unbound variable

shr-project committed:
Jenkins/check-unity-log.sh: check also for 'Player build result: Failed!'

* instead of detecting failed build just because of missing output binary

kamkolak-lge committed:
Revert FBX package to 3.2.1-preview.2

shr-project committed:
Jenkins: use local UNITY_DOCKER_IMAGE

* instead of global configure variable from Jenkins

KarByc committed:
[Network] Improved load balancing of the sensors.

heeen committed:
[FEATURE] [AUTO-6237] SDF import

shr-project committed:
Jenkins: set UNITY_VERSION and check Unity Editor existence

EricBoiseLGSVL committed:
fix null after version is wrong

Commit:
[FEATURE] [AUTO-5509] Add proxy support for simulator cloud integration

KarByc committed:
Simulator ignores IP addresses of the inactive interfaces.

kamkolak-lge committed:
[Feature] [AUTO-6139] [AUTO-6427] Bridge thread pool

heeen committed:
SDF: support link self collision, initial plugin support

shr-project committed:
Jenkins: use hdrp/unityci/editor docker image

* built in:
  https://auto-jenkins.lgsvl.net/view/Jenkins%20test/job/unityci-docker-image/12/console
  but
  https://hub.docker.com/layers/unityci/editor/ubuntu-2020.3.3f1-windows-mono-0.13.0/images/sha256-3eaa8da58fda00f5f198527339b5839a46f3af6f3f0eca8e8f1f4a0510665f64?context=explore
  https://hub.docker.com/layers/unityci/editor/ubuntu-2020.3.3f1-mac-mono-0.13.0/images/sha256-9ff06baedd8ff428d6151c36ea91494d70058fa2282d1fd458dcff6b1021f1cd?context=explore
  https://hub.docker.com/layers/unityci/editor/ubuntu-2020.3.3f1-android-0.13.0/images/sha256-c8f65a43f7436ccd082e658a9b683f0db5c81ea3c71e764e3ce886e6e9743e5e?context=explore
  should work as base image as well (just different base image for each platform until
  https://github.com/game-ci/docker/pull/116
  is merged upstream and some image with combination of platforms we use is on dockerhub.

KarByc committed:
[Network] [AUTO-6586] Distributed simulation fixes

KarByc committed:
[Network] Increased margin for the map size bounds.

EricBoiseLGSVL committed:
[FIX] [AUTO-6698] Sentry fixes

EricBoiseLGSVL committed:
[FEATURE] [AUTO-6850] [AUTO-6852] Robotics support

kamkolak-lge committed:
Fix initialization order, use main thread for time

kamkolak-lge committed:
[Fix] [AUTO-6766] Fix crash in THREE.js loader in WISE for glTF files with TGA or TIF textures

EricBoiseLGSVL committed:
update map and vehicle version error message

KarByc committed:
[Sensors] [AUTO-6681] Sensors rework

heeen committed:
Differential drive logic from cloisim

kamkolak-lge committed:
[Fix] [AUTO-7028] Improve data handling in HD map mesh builder

kamkolak-lge committed:
Remove overdraw with HDRP flare

KarByc committed:
[VSE] [-] Minor UX fixes.

mark-gerow-lge committed:
[FEATURE] [AUTO-7018] Developer Debug Mode

mark-gerow-lge committed:
Asmdef added, loader fixed to deal with bundle/WISE name mismatch

KarByc committed:
[Network] [AUTO-6969] Support for the ArticulationBody in distributed simulations

shr-project committed:
Loader: improve the error message about BundleFormat

* there was missing space before "for Lincoln...":
  Exception: BundleFormat version com.svlsimulator.4 mismatch expected com.svlsimulator.5for Lincoln2017MKZ eef433df-fd77-41ec-ad6a-0080e05b5428

* re-phrase it, so it's more clear that in this case it was the
  Lincoln2017MKZ asset being too old with version com.svlsimulator.4
  and the Simulator (svlsimulator-linux64-master-1487 too new for that)
  not the other way around as how the exception text sounded (at least
  to me and Keith)

lokesku committed:
Add note to highlight diff between release vs master branch for builds

EricBoiseLGSVL committed:
fix incorrect raycast and addagent rigidbody null check

KarByc committed:
[Network] [AUTO-7103] Improved distributing VehicleActions.

KarByc committed:
[Network] Fixing Sensors initialization.

heeen committed:
Sensor debug developer loading

EricBoiseLGSVL committed:
fix missing editor check

shr-project committed:
Jenkins: add some explanation about the EDITOR_DOCKER_IMAGE and how to build it locally

EricBoiseLGSVL committed:
[Feature] [AUTO-7132] Add simulations to cache clear options

eagafonovlge committed:
Jenkins: Change customWorkspace for unityci-based images

shr-project committed:
Jenkins: use more generic gpu-vulkan label

* I've updated auto-jenkins builders to use this, instead of a bit confusing
  unity3d-2019.4 prefix as we're using 2020.3.3f1 version now

daviduhm committed:
Take MapOrigin transform into account for northing and easting conversion

KarByc committed:
[Core] Fixed rotating NPC wheels when using NPCWaypointBehaviour.

kamkolak-lge committed:
[Feature] [AUTO-7083] Erosion and re-leveling for point cloud mesh

mark-gerow-lge committed:
[FEATURE] [AUTO-7188] Added datetime to asset model

EricBoiseLGSVL committed:
add steer angle to waypoint follow

eagafonovlge committed:
Update default cloud_url

Use wise.preview.svlsimulator.com for testing the bleeding-edge
simulator

KarByc committed:
Added Houdini Engine Unity files to the gitignore.

heeen committed:
wait a bit between trying master endpoint adresses

Commit:
Tool that helps you set the accurate values (position, rotation and scale) for MapOrigin

EricBoiseLGSVL committed:
Fix issue with agent dropdown not refreshing on click

kamkolak-lge committed:
[Fix] Fix TF visualizer for sensors

heeen committed:
AUTO-7213 set stopping state regardless of online state

kamkolak-lge committed:
[Fix] [AUTO-7134] [AUTO-7258] Cubemap support for point cloud + fixes

mark-gerow-lge committed:
Fix db syntax error

KarByc committed:
[Core] Trimming last slash in the cloud url if needed.

mark-gerow-lge committed:
Bridge Support integration

mark-gerow-lge committed:
[AUTO-6592] List supported bridge types in manifest

mark-gerow-lge committed:
Bundle version uptick and unity editor check on developer debug mode for bridges

heeen committed:
do not perform stop sequence when already idle

kamkolak-lge committed:
Change method signatures for backward compatibility

eagafonovlge committed:
Jenkinsfile: Checkout and build bridges

mark-gerow-lge committed:
Custom attributes moved to core, getter for bridgenameattribute

eagafonovlge committed:
Fix sensor build

eagafonovlge committed:
Make sure Simulator.Editor.Build.Run exits on any exception

kamkolak-lge committed:
[Feature] [AUTO-7130] Rain collision

kamkolak-lge committed:
Rain collision improvements and fixes

heeen committed:
[FIX] [-] Rework developer settings

KarByc committed:
[Network] [AUTO-7265] Fixed API mode for distributed simulation on a single machine.

ericSterner committed:
[FIX] [-] Reflection probe and preview fixes

EricBoiseLGSVL committed:
enable light layer option in hdrp options

mark-gerow-lge committed:
[Auto-6354] Preview manifest fix

mark-gerow-lge committed:
API mode add agent fix

EricBoiseLGSVL committed:
fix network type check typo

EricBoiseLGSVL committed:
[FIX] [-] Fix for network check on non distributed

KarByc committed:
[Network] Fixed pedestrian animations on client machine.

lokesku committed:
fixup! [AUTO-6410] Update Unity to 2020.3.3f1

lokesku committed:
fixup! Add new FORCE_DEV_BUILD param to jenkins

EricBoiseLGSVL committed:
[FEATURE] [-] Pedestrian ground truth from bounds

mark-gerow-lge committed:
[FIX] [AUTO-7335] [AUTO-7362] Bridge Load Fix

mark-gerow-lge committed:
Assembly Load Fix

EricBoiseLGSVL committed:
api set sim camera state

kamkolak-lge committed:
Support default light layers in point cloud renderer

heeen committed:
fix trying to load empty bridges assembly

mark-gerow-lge committed:
Bridge build fix

EricBoiseLGSVL committed:
[FIX] [-] Api rain fix

EricBoiseLGSVL committed:
update unity bundle version to 2021.2

shr-project committed:
Assets: add assert for null assetGuid and show its type

* will show at least the type of asset which was null instead of
  completely useless error shown in exception:

  PLC_B_15 - Traceback (most recent call last):
    File "/OE/projects/lgsvl/HDRP/PythonApi/examples/NHTSA/Perform-Lane-Change/PLC_B_15.py", line 44, in <module>
      ego = sim.add_agent(env.str("LGSVL__VEHICLE_0", lgsvl.wise.DefaultAssets.ego_lincoln2017mkz_apollo5_modular), lgsvl.AgentType.EGO, egoState)
    File "/OE/projects/lgsvl/HDRP/PythonApi/lgsvl/utils.py", line 23, in new_f
      return f(*args, **kwargs)
    File "/OE/projects/lgsvl/HDRP/PythonApi/lgsvl/simulator.py", line 162, in add_agent
      uid = self.remote.command("simulator/add_agent", args)
    File "/OE/projects/lgsvl/HDRP/PythonApi/lgsvl/remote.py", line 69, in command
      raise Exception(data["error"])
  Exception: Value cannot be null.
  Parameter name: path2

* the Player.log exception with a stack trace shows at least some context for this
  error, but still doesn't show which type of asset it was (so it's still difficult
  to guess which assetGuid was null (Bridge in this case):

  Failed to start 'Random' simulation

  ArgumentNullException: Value cannot be null.
  Parameter name: path2
    at System.IO.Path.Combine (System.String path1, System.String path2) [0x00011] in <eae584ce26bc40229c1b1aa476bfa589>:0
    at Simulator.Web.WebUtilities.GenerateLocalPath (System.String assetGuid, Simulator.BundleConfig+BundleTypes type) [0x00020] in <de43adacb40d4d31837e9d3d6facd55d>:0
    at Simulator.Web.DownloadManager.GetAsset (Simulator.BundleConfig+BundleTypes type, System.String assetGuid, System.String name, System.IProgress`1[T] progressCallback) [0x0009b] in <de43adacb40d4d31837e9d3d6facd55d>:0
    at Simulator.Loader+<StartSimulation>d__43.MoveNext () [0x002da] in <de43adacb40d4d31837e9d3d6facd55d>:0
  UnityEngine.DebugLogHandler:Internal_LogException(Exception, Object)
  UnityEngine.DebugLogHandler:LogException(Exception, Object)
  UnityEngine.Logger:LogException(Exception, Object)
  UnityEngine.Debug:LogException(Exception)
  Simulator.<StartSimulation>d__43:MoveNext()
  System.Runtime.CompilerServices.AsyncVoidMethodBuilder:Start(<StartSimulation>d__43&)
  Simulator.Loader:StartSimulation(SimulationData)
  <>c__DisplayClass18_1:<Parse>b__2()
  ConnectionManager:Update()

  loader status: Loading->Error Value cannot be null.
  Parameter name: path2

* with this change applied it's at least showing the type of asset which is null:

  PLC_B_15 - Traceback (most recent call last):
    File "/OE/projects/lgsvl/HDRP/PythonApi/examples/NHTSA/Perform-Lane-Change/PLC_B_15.py", line 44, in <module>
      ego = sim.add_agent(env.str("LGSVL__VEHICLE_0", lgsvl.wise.DefaultAssets.ego_lincoln2017mkz_apollo5_modular), lgsvl.AgentType.EGO, egoState)
    File "/OE/projects/lgsvl/HDRP/PythonApi/lgsvl/utils.py", line 23, in new_f
      return f(*args, **kwargs)
    File "/OE/projects/lgsvl/HDRP/PythonApi/lgsvl/simulator.py", line 162, in add_agent
      uid = self.remote.command("simulator/add_agent", args)
    File "/OE/projects/lgsvl/HDRP/PythonApi/lgsvl/remote.py", line 69, in command
      raise Exception(data["error"])
  Exception: assetGuid must not be null when trying to get $Bridge asset.
  Assertion failure. Value was Null
  Expected: Value was not Null

heeen committed:
AUTO-6983 do not reconnect bridge in error or stopping state

KarByc committed:
[VSE] Catching initialization exceptions.

shr-project committed:
ScenarioManager: don't try to set non-existent IsInitializing property

* it exists only in vse branch where it was added with this commit:
  https://auto-gitlab.lgsvl.net/HDRP/Simulator/commit/08038b7a2b87e1608aebe757e74d6acd774006fb#b8a2054485aa4aa693a454077a9753bb5fed45cc
  but it doesn't exist in master causing:
  04:10:58  Assets/Scripts/ScenarioEditor/Managers/ScenarioManager.cs(310,13): error CS0103: The name 'IsInitializing' does not exist in the current context

* and assign to private isInitialized property directly to fix:
  04:10:58  Assets/Scripts/ScenarioEditor/Managers/ScenarioManager.cs(311,13): error CS0200: Property or indexer 'ScenarioManager.IsInitialized' cannot be assigned to -- it is read only

EricBoiseLGSVL committed:
[FEATURE MAPS] [-] Annotation size

heeen committed:
handle api disconnect causes expected task cancellation

KarByc committed:
[VSE] [AUTO-7183] Support for the BezierSpline paths in the waypoints.

ericSterner committed:
Changes to sun to accomodate proper lighting

KarByc committed:
[VSE] [-] Fixes for the VSE

heeen committed:
Only load requested bridge, add logging around loading of plugin sources

heeen committed:
update websocket sharp, use current source instead of dll

mark-gerow-lge committed:
Initial commit

heeen committed:
fix empty buildversion

heeen committed:
new task progress interface

This reverts commit 733aff096b2b57e6175ae81a250690b8e7479150.

heeen committed:
[FEATURE NPC] [AUTO-7253] NPCManager extention

kamkolak-lge committed:
Sensor related changes

- Sensor visualizer can now support multiple feeds
- Added support for uncompressed image message format
- Bridge message dispatcher now support sub-pools for each message format

kamkolak-lge committed:
Add LaserScan sensor data type

heeen committed:
[REFACTOR] Make Loader methods non-static

shr-project committed:
Jenkins: pin aws-cli version to 2.2.20

* to match what other HDRP projects are using instead of using
  whatever is latest when running the build, last simulator
  build from master:
  https://auto-jenkins.lgsvl.net/job/simulator/job/master/1582/
  was already using 2.2.26, but downgrade back to 2.2.20 is
  OK and then we'll upgrade all projects at the same time in
  controlled way (there is new aws-cli release almost every
  day, but we don't need to update so often).

* changes:
  https://github.com/aws/aws-cli/compare/2.2.20...2.2.26

kamkolak-lge committed:
[Fix] [AUTO-7391] Point cloud fixes

EricBoiseLGSVL committed:
development to version support

EricBoiseLGSVL committed:
create directory before lock file

Commit:
[Fix][AUTO-6618] Added TargetGear in for Autoware Auto reverse driving.

EricBoiseLGSVL committed:
version uptick and application version

EricBoiseLGSVL committed:
Add checks for pedestrian and traffic annotation for api random agents

EricBoiseLGSVL committed:
pass position and rotation to SpawnAgent in config

kamkolak-lge committed:
Redirect stdout and stderr from FBX2glTF

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/lgsvl/simulator?pullRequest=1635) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/lgsvl/simulator?pullRequest=1635) before we can accept your contribution.<br/>**0** out of **13** committers have signed the CLA.<br/><br/>:x: KarByc<br/>:x: ImaginePlay<br/>:x: heeen<br/>:x: RLitynskyy<br/>:x: EricBoiseLGSVL<br/>:x: daviduhm<br/>:x: ericSterner<br/>:x: kamkolak-lge<br/>:x: mark-gerow-lge<br/>:x: lemketron<br/>:x: hadiTab<br/>:x: lokesku<br/>:x: shr-project<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/lgsvl/simulator?pullRequest=1635) it.</sub>

kamkolak-lge committed:
Fix broken assembly loads when bundles have no scripts

pijaro committed:
Get bridge by name, not a type

Signed-off-by: Piotr Jaroszek <piotr.jaroszek@robotec.ai>

shr-project committed:
docker-compose: use latest hdrp/unityci/editor docker image

* built in:
  https://auto-jenkins.lgsvl.net/view/docker/job/unityci-docker-image/20/console

shr-project committed:
build-simulator.sh: show which simulator executable wasn't found

EricBoiseLGSVL committed:
npc support independent of size, map preview option

EricBoiseLGSVL committed:
[FEATURE] [-] Ego lane follow dynamics

EricBoiseLGSVL committed:
lane follow SMI clean up

heeen committed:
Add Version Selection to DevelopmentSettings, fixes

KarByc committed:
[Fix] Setting name and assetguid in the NPC and pedestrian managers.

EricBoiseLGSVL committed:
[FEATURE] [AUTO-7423] Import hdmap refactor

EricBoiseLGSVL committed:
Get baselink transform on build

heeen committed:
SDF: allow selection of model path

refactor to split into more classes/files
check orientation and size of collision is correct

mark-gerow-lge committed:
[AUTO-7569] First Pass UI Rework for Loaded Assets and other QOL

kamkolak-lge committed:
[Feature] [AUTO-7549] Rework perception sensor 2D to use image data

EricBoiseLGSVL committed:
uptick vehicle version to align with next release

heeen committed:
add stl parser

heeen committed:
Docker support, add Console

kamkolak-lge committed:
Fix camera stutter when physics is paused

mark-gerow-lge committed:
[AUTO-7182] Do not show popup on failed build

kamkolak-lge committed:
Fix segmentation id when NPC is spawned through Python API

mark-gerow-lge committed:
Correctly display sim version

heeen committed:
use coroutine for bridge client connection watchdog

heeen committed:
fix error reporting from bridge client

mark-gerow-lge committed:
Null Check on DevSettings

RLitynskyy committed:
[FIX OPENDRIVE] [AUTO-6711] Fixed OpenDrive MapOrigin export logic

kamkolak-lge committed:
[Fix] [AUTO-7599] Fix `export point cloud` option

RLitynskyy committed:
Added check for traffic lane info presence to improve error handling.
Some comments and minor code refactor.

RLitynskyy committed:
[Fix] [AUTO-6491] Lane, Sign and Signal IDs import & export

kamkolak-lge committed:
Fix possible missing key exception

kamkolak-lge committed:
Add async GPU readback pool

eagafonovlge committed:
AUTO-6872: Update docker image handling

- Do not use URI since a docker image is not an URI
- Get rid of separated registry and image in favor of single
registryImage
- Update debug logging a bit

eagafonovlge committed:
AUTO-6872: Adopt docker terminology

Docker Registry VS Repository VS Image VS Tag

https://developers.redhat.com/blog/2016/01/13/a-practical-introduction-to-docker-container-terminology#repository

TL;DR:
1. Image is Repository + Tag
2. Same string could be either a repository or an image implying
`latest` tag

Example:

Docker Hub's `amazon/aws-cli:latest` is an image where `amazon/aws-cli`
is the Repository and `latest` is the Tag

```
minikube:30080/svlcontent/testcaserunner/pythonapi:v42
<- registry ->|
<---------- repository -------------------------->|tag
```

The Registry is a hosting service for Repositories.
Docker Hub is just a registry that is used by default.

KarByc committed:
[VSE] [AUTO-7412] VSE improvements

daviduhm committed:
[AUTO-7425] Add an API command for canceling destination

shr-project committed:
Jenkins: build each bundle in separate Unity invocation

* Unity now fails when building many bundles at the same time
  so many separate jenkins jobs needs to be triggered to build
  them one-by-one, change build-bundles.sh to build them separately
  there, this way all the assets can be checked out by the
  Jenkins job and then only the Unity build is invoked separately

EricBoiseLGSVL committed:
adding a few missed metas and update package manifest

heeen committed:
use special hostname to connect to windows docker host

KarByc committed:
[VSE] Fix for editing controllables policy and resetting map signals policies.

daviduhm committed:
[AUTO-7690] Implement API function to set NavOrigin in scene

shr-project committed:
Jenkins: get-assets.sh: show a bit more debug output

* S3_DOWNLOAD_HOST variable was still pointing to assets.dev.lgsvlsimulator.com
  which was shutdown on July 18th and now all asset bundle builds were failing
  without any useful output when FORCE_REBUILD wasn't enabled, because this script
  runs with "set -e" and curl was returning error:
  6      Couldn't resolve host. The given remote host was not resolved.
  so the Jenkins build suddenly failed without any output:

  12:13:24   > git lfs pull origin # timeout=10
  12:13:25  Commit message: "update mats"
  [Pipeline] echo
  12:13:25  INFO: Checked out following Environments: CubeTown,...
  [Pipeline] sh
  12:13:25   > git rev-list --no-walk e030b652049c9ae0af4978d7cdf3d64933167c7f # timeout=10
  12:13:25  SUBFOLDERS=CubeTown ...
  [Pipeline] }
  [Pipeline] // script
  [Pipeline] }
  [Pipeline] // stage
  [Pipeline] stage
  [Pipeline] { (Checkout Vehicles)
  Stage "Checkout Vehicles" skipped due to earlier failure(s)

  only after adding -x to get-assets it was showing reasonable output to figure out
  what went wrong:

  12:58:48  ++ basename Assets/External/Environments/CubeTown
  12:58:48  + NAME=CubeTown
  12:58:48  + [[ ! CubeTown =~ @tmp ]]
  12:58:48  ++ GIT_DIR=Assets/External/Environments/CubeTown/.git
  12:58:48  ++ git rev-parse HEAD
  12:58:48  + ID=d1acb0b59326407c91906cb3a0a3901286adc984
  12:58:48  + '[' 0 -eq 1 ']'
  12:58:48  ++ curl -sILw '%{http_code}\n' https://assets.dev.lgsvlsimulator.com/v2/d1acb0b59326407c91906cb3a0a3901286adc984/environment_CubeTown -o /dev/null
  12:58:48  + CHECK=000
  [Pipeline] }
  [Pipeline] // script
  [Pipeline] }
  [Pipeline] // stage
  [Pipeline] stage
  [Pipeline] { (Checkout Vehicles)
  Stage "Checkout Vehicles" skipped due to earlier failure(s)

  include enough output to show what this script is doing during the build
  (without all the clutter from -x).

shr-project committed:
Jenkins: build-bundles.sh: use 4 spaces for indentation

* it was using mix of 2 and 4 spaces

shr-project committed:
Jenkins: Checkout bundle sources to .external-assets

* to prevent Unity importing all the available assets when we want to build just one of them
* then inside the for loop in build-bundles.sh always move the source directory from
  .external-assets to External and move it back after the build
* also check if the Asset bundle was created in expected location
* this also helps when there are many Assets selected in Jenkins
  job parameters, but get-assets.sh filters only very few of them
  to be built (without FORCE_REBUILD being used)

shr-project committed:
Jenkins: use latest hdrp/unityci/editor docker image

daviduhm committed:
[AUTO-7691] Add API call to get NavOrigin from scene

KarByc committed:
Extracted Pedestrians' behaviours from the PedestrianController.

Fixed time to collision effector.

RLitynskyy committed:
[Fix] [AUTO-7315] Fix OpenDrive1.4 "michigan intersection" map crash

kamkolak-lge committed:
Add bounding box calculation for segmentation map

shr-project committed:
AUTO-5652: Jenkins: Remove scenario runner from Simulator zipfile

* rename the "*-linux64-${OUTPUT_SUFFIX}-norunner.zip" back to normal
  "*-linux64-${OUTPUT_SUFFIX}.zip" and remove whole bundleScenariosRunner
  step

* drop SCENARIOS_RUNNER_DOCKER_TAG job parameter

* update UNITY_DOCKER_IMAGE_TAG in release job to match normal job
  and bundle jobs

* use env.OUTPUT_SUFFIX in normal job as release job was using since
  5d3a47dec90886706e9d8d69159aba304673e93c

Commit:
[FEATURE Settings] [AUTO-7668] Enable Settings menu in loader scene

RLitynskyy committed:
[fix OpenDrive] [AUTO-7777] Fix OpenDrive map export and build

mark-gerow-lge committed:
[FIX] [-] Refresh cached sims on delete asset

EricBoiseLGSVL committed:
Remove old input unity work arounds for Linux

Commit:
Custom managers in NPC behaviours

EricBoiseLGSVL committed:
[UPDATE Sentry] [AUTO-7735] Add simulator version to sentry event extra details and remove screen orientation

KarByc committed:
[Core] NPC/Pedestrian pool no longer removes existing agents.

NPCs and Pedestrians gets despawned before being destroyed.
SimulatorManager creates new SegmentationIdMapping on simulation init.

heeen committed:
DockerDotnet modifications

Modifications to DockerDotnet source

heeen committed:
add docker dotnet source

Replace Docker Dotnet assembly with source checkout:

git@github.com:dotnet/Docker.DotNet.git
e51de97addebec71c1b1dd7f399f06e381a1044c

remove tests

heeen committed:
fix named pipes on windows

named pipes in .net on windows block writing until a previous read has
finished. Use peek to check if anything can be read, and if not back off
for 100ms, during which we can write.

heeen committed:
Fix NPC assembly could be empty

also fix development mode bridge should be null if no bridge can be
selected for local prefab

EricBoiseLGSVL committed:
remove missed meta files

Commit:
[FEATURE] [AUTO-7719] Light layer changes should be automatic

shr-project committed:
Jenkins: Fix unityci based docker image

* switches from ubuntu-18.04 to ubuntu-20.04 to fix vulkan renderer

* install libvulkan1 and remove ca-certificates
* update the comments

* we're using -force-vulkan in our builds

* install libvulkan1 as it was before the switch to unityci docker image
  because without this Unity will fail to detect Vulkan and use llvmpipe
  renderer instead, which results in all black preview images in Assets
  bundles and we also use -force-vulkan, so it's a bit surprising that
  missing libvulkan1 wasn't fatal error

* without mesa-vulkan-drivers vulkaninfo would fail with:
  root@0ea8660cc34a:/# vulkaninfo
  ERROR: [Loader Message] Code 0 : loader_scanned_icd_add: Could not get 'vkCreateInstance' via 'vk_icdGetInstanceProcAddr' for ICD libGLX_nvidia.so.0
  Cannot create Vulkan instance.
  This problem is often caused by a faulty installation of the Vulkan driver or attempting to use a GPU that does not support Vulkan.
  /build/vulkan-tools-KEbD_A/vulkan-tools-1.2.131.1+dfsg1/vulkaninfo/vulkaninfo.h:371: failed with ERROR_INCOMPATIBLE_DRIVER

  but having mesa-vulkan-drivers in the image breaks Unity trying to use Vulkan

* I don't see ca-certificates being used

* use latest hdrp/unityci/editor docker image built in:
  https://auto-jenkins.lgsvl.net/view/docker/job/unityci-docker-image/28/

* don't use xvfb-run

* update image-info-lgsvl.source
  when starting from editor-lgsvl instead of regular editor from Docker Hub
  the /opt/unity/image-info-lgsvl.source would already be there
  but we don't overwrite it, instead add another section after
  IMAGE_APP="unity-editor-simulator"
  showing few more details, but drop UNITY_CHANGESET, UNITY_MODULES
  params as they are never set when building ${UNITY_DOCKER_IMAGE_TAG}-simulator
  image and we "inherit" them from editor-lgsvl image used in EDITOR_DOCKER_IMAGE

shr-project committed:
Jenkins: show some output about signing the windows64 executable

* otherwise it shows just "Succeeded" or "Failed" in the log
  without any context to guess that it's from osslsigncode

daviduhm committed:
[AUTO-7325] [AUTO-7340] Add python API commands for Destination sensor

heeen committed:
improve SDF import dialog

KarByc committed:
[Fix] Fixed NPC controller stopping all coroutines on the initialization.

ImaginePlay committed:
Moved input actions to sensor bundles.

heeen committed:
correct name of cyber bridge asmdef file

ImaginePlay committed:
Fix for Input package error on lost focus.

KarByc committed:
[FEATURE VSE] [AUTO-7588] VSE support for the NavOrigin.

KarByc committed:
Local variable name correction.

KarByc committed:
[VSE] Enabled NavOrigin rotation.

RLitynskyy committed:
[UPDATE HDMAP] [AUTO-6928] Remove Apollo 3.0 support

mark-gerow-lge committed:
[FEATURE] [-] API versioning support

KarByc committed:
[Fix] Fixed initializing spawns manager if there is no active agents.

ericSterner committed:
Adding detail map and albedo tint to environment shader

daviduhm committed:
Remove apollo 3.0 from manifest

EricBoiseLGSVL committed:
[FEATURE] [AUTO-7883] Map origin rework

heeen committed:
[FIX DOWNLOAD] [AUTO-7783]  fix some invalid state transitions

shr-project committed:
Jenkins: build-bundles: retry building each bundle upto 3 times

* when rebuilding CubeTown many times in the same workspace I always get segfault like:

  19:06:26  Launched and connected shader compiler UnityShaderCompiler after 0.03 seconds
  19:06:44  Caught fatal signal - signo:11 code:1 errno:0 addr:0xd0000000e
  19:06:44  Obtained 25 stack frames.
  19:06:44  #0  0x007f72a328a3c0 in funlockfile
  19:06:44  #1  0x007f72a3141c5b in memcpy
  19:06:44  #2  0x007f71edcabf64 in vk_optimusGetDeviceProcAddr
  19:06:44  #3  0x0056107c6dad75 in vk::TaskExecutor::HandleCommandStream(GrowableBuffer*, VkCommandBuffer_T*)
  19:06:44  #4  0x0056107c6d9fe6 in vk::TaskExecutor::ExecuteCommandbuffers(vk::CommandBuffer**, int)
  19:06:44  #5  0x0056107c6d6c33 in vk::TaskExecutor::DoFlush(VkFence_T*, VkSemaphore_T*, unsigned long)
  19:06:44  #6  0x0056107c6a3e07 in GfxDeviceVK::SubmitCurrentCommandBuffers(VkSemaphore_T*)
  19:06:44  #7  0x0056107c6a5721 in GfxDeviceVK::FinishRendering()
  19:06:44  #8  0x0056107c61f4dd in GfxDeviceClient::FinishRendering()
  19:06:44  #9  0x0056107bf9b506 in AssetHotreload()
  19:06:44  #10 0x0056107f0fb8ff in AssetHotreload(core::hash_set<UnityGUID, core::hash<UnityGUID>, std::equal_to<UnityGUID> > const&)
  19:06:44  #11 0x0056107f1964e7 in RefreshInternalV2(AssetDatabase::UpdateAssetOptions, ScanFilter const&, InternalRefreshFlagsV2)
  19:06:44  #12 0x0056107f180712 in StopAssetImportingV2(AssetDatabase::UpdateAssetOptions, InternalRefreshFlagsV2, ScanFilter const*)
  19:06:44  #13 0x0056107f180cb1 in RefreshV2(AssetDatabase::UpdateAssetOptions)
  19:06:44  #14 0x0056107f0e5181 in AssetDatabase::Refresh(AssetDatabase::UpdateAssetOptions)
  19:06:44  #15 0x0056107e4fdeb3 in Application::InitializeProject()
  19:06:44  #16 0x0056107f03e606 in InitializeUnity(void*)
  19:06:44  #17 0x007f72a39c3be8 in g_source_get_time
  19:06:44  #18 0x007f72a39c304e in g_main_context_dispatch
  19:06:44  #19 0x007f72a39c3400 in g_main_context_dispatch
  19:06:44  #20 0x007f72a39c36f3 in g_main_loop_run
  19:06:44  #21 0x007f72a405837d in gtk_main
  19:06:44  #22 0x0056107f03dfce in main
  19:06:44  #23 0x007f72a30aa0b3 in __libc_start_main
  19:06:44  #24 0x0056107bd72029 in _start
  19:06:44  /usr/bin/unity-editor: line 11:    11 Segmentation fault      (core dumped) /opt/unity/Editor/Unity "$@"

  when building in workspace with empty "Library" directory and re-running
  it in the same workspace 2nd time (keeping "Library") finishes OK

  Rerun each bundle build upto 3 times before giving up and showing an error.

  Show summary at the end to easily find if some bundles still need to be
  re-triggered or fixed.

* re-try the initial "fetch license" as well as it was sometimes failing
  there already (before starting any bundle build)

* we do something similar in build-simulator.sh already, for "check" function

kamkolak-lge committed:
Fix lighting on API agent spawnt

shr-project committed:
Jenkins: remove some unnecessary changes

* now when Unity doesn't segfault when Library is empty we can remove some work arounds
  which seemd to fix some of the segfault causes

* we still cannot use Xvfb, it still segfaults even when repeating build 3 times and
  keeping Library between the builds

* use latest hdrp/unityci/editor docker image built in:
  https://auto-jenkins.lgsvl.net/view/docker/job/unityci-docker-image/30/

shr-project committed:
Jenkins: upload-bundles: allow public-read for development S3 bucket

KarByc committed:
[Pedestrians] [AUTO-7842][AUTO-7924] Removed NavSystem from pedestrian waypoint behaviour.

shr-project committed:
Jenkins: fix unbound S3_BUCKET_ACL

* previous MR was merged so fast that the test build didn't even
  start before that, fix unbound S3_BUCKET_ACL and counting of BUNDLEs
* fix shebangs in upload-bundles*.sh to bash as other scripts are
  using, we have (useless) "function" bashism inside, so lets not
  claim /bin/sh compatiblity

heeen committed:
fix bridgeclient connection with null connection string

KarByc committed:
[VSE] Fixed playback mode with triggers.

KarByc committed:
[VSE] Unified scenario json parameters' casing.

mark-gerow-lge committed:
[FIX API] [AUTO-7927] LoadScene API Fix

heeen committed:
[FIX BRIDGE] [AUTO-6983] Fix Bridge connection watchdog

EricBoiseLGSVL committed:
Support scrolling for loaded assets

KarByc committed:
[Fix] Fixed agents rotation in follow waypoints commands.

Fixed loading scenes in the Python API.

kamkolak-lge committed:
[Fix] [AUTO-7922] Fix Python API indoor lighting issues

heeen committed:
[UI/Loader] [AUTO-7469] update ui during connection attempts

heeen committed:
further improve connection UI

hadiTab committed:
Check if destination info object is valid

EricBoiseLGSVL committed:
hack to disbale dev console when sim console is active

daviduhm committed:
Revert "[AUTO-7425] Add an API command for canceling destination"

This reverts commit 465655595d0d6feb08a419b5963364ed55dba849.

EricBoiseLGSVL committed:
[FIX] [-] Check if annotation is found before setting API walk randomly

lokesku committed:
AUTO-7880: Update 3rd party OSS list

kamkolak-lge committed:
Remove debug log

KarByc committed:
[Fix] Unloads asset bundles after importing config files.

EricBoiseLGSVL committed:
[FIX] [-] Fix bundle texture stream issue

hadiTab committed:
AUTO-7943: Add Python API command for getting bridge type

KarByc committed:
[Fix] Fixed load map method when there is no callback passed.

Commit:
Fixes GPS and CANBUS sensor type names usage

EricBoiseLGSVL committed:
[UPDATE] [-] Update docker version

KarByc committed:
[Fix] Fixed caching controllables' assets in dev mode.

kamkolak-lge committed:
[AUTO-7950] Update Unity to 2020.3.19f1 and HDRP to 10.7.0

EricBoiseLGSVL committed:
Fix f1tenth npc light layer trigger size

kamkolak-lge committed:
Fix ego detection in 2D perception sensor

lemketron committed:
[UPDATE] Add Sunset info to README

evojinic commented:
Cruise
